blockchain/network.py

```diff
@@ -678,6 +794,40 @@ def receipt_request(self, pass_phrase):
             transfer_node = self.registered_nodes[pass_phrase]
             return self.get_unsent_transfer_ids(transfer_node.node_id)
 
+    def subscription_provisioning(self, subscription_id, criteria, phase_criteria, public_key):
+        """ initial communication between subscription """
+        try:
+            subscribe_from_db.insert_subscription(subscription_id, criteria, phase_criteria, public_key)
+        except:
+            logger().warning("A subscription SQL error has occurred.")
+        pass
+
+    def subscription_request(self, subscription_id, subscription_signature):
+        """ return transactions and associated verification records that meet criteria made by client """
+        transactions = []
+        verification_records = []
+        subscriber_info = subscribe_from_db.get(subscription_id)
+        # convert thrift signature to dictionary
+        subscriber_signature = thrift_converter.convert_thrift_signature(subscription_signature)
+        criteria = subscriber_info['criteria']
+        public_key = subscriber_signature["public_key"]
+        # validate subscription signature
+        if validate_subscription(subscriber_signature, criteria, public_key):
+            # query transactions/vrs ready to send to calling subscription node
+            try:
+                subscription_messages = list(sub_vr_transfers_db.get_all(subscriber_info['subscriber_id']))
```

Comment:
The server will pass back a correlation id to the client. The client will call with the most recent correlation id. At which point the server will blow anything away matching that correlation id and respond with everything else.